lib/modules/hosts/youtube.js

```diff
@@ -11,49 +12,48 @@ export default {
 			(/^https?:\/\/(?:youtu\.be|(?:www\.|m\.)?youtube\.com)\/?(?:watch|embed)?(?:.*?v=|v\/|\/)([\w-]+)/i).exec(href);
 	},
 	handleLink(elem, [, hash]) {
-		// Check url for timecode e.g t=1h23m15s
-		const starttimeRe = /[&\?](?:t=|start=)([0-9hms]+)/i;
-		const starttimeResult = starttimeRe.exec(elem.href);
-		// End time is always in integer format, no need to parse
-		const endtimeRe = /[&\?]end=([0-9]+)/i;
-		const endtimeResult = endtimeRe.exec(elem.href);
-		// Finds playlist id if included
-		const playlistRe = /[&\?]list=([\w\-_]+)/i;
-		const playlistResult = playlistRe.exec(elem.href);
+		let src = `https://www.youtube.com/embed/${hash}?enablejsapi=1&enablecastapi=1`;
+		const paramResults = getUrlParams(elem.href);
+		for (const paramName in paramResults) {
```

Comment:
Seems like `if (params.t)`, `if (params.end)`, etc. would be simpler than looping with a switch